File: sdks/python/freemail/resources/pods.py
from __future__ import annotations

from typing import TYPE_CHECKING, Optional

if TYPE_CHECKING:
    import httpx


class PodResource:
    """Operations on pods."""

    def __init__(self, client: httpx.Client) -> None:
        self._client = client

    def _request(self, method: str, path: str, **kwargs) -> dict:
        from ..client import _handle_response
        resp = self._client.request(method, path, **kwargs)
        return _handle_response(resp)

    def list(
        self,
        limit: int = 25,
        page_token: Optional[str] = None,
    ) -> dict:
        params: dict = {"limit": limit}
        if page_token:
            params["page_token"] = page_token
        return self._request("GET", "/pods", params=params)

    def get(self, pod_id: str) -> dict:
        return self._request("GET", f"/pods/{pod_id}")

    def create(
        self,
        name: str,
        description: Optional[str] = None,
    ) -> dict:
        body: dict = {"name": name}
        if description is not None:
            body["description"] = description
        return self._request("POST", "/pods", json=body)

    def delete(self, pod_id: str) -> None:
        self._request("DELETE", f"/pods/{pod_id}")

File: sdks/node/src/resources/pods.ts
import type {
  Pod,
  PaginatedResponse,
  CreatePodOptions,
  ListPodOptions,
} from "../types.js";

type RequestFn = <T>(method: string, path: string, body?: unknown) => Promise<T>;

export class PodResource {
  constructor(private request: RequestFn) {}

  async list(options?: ListPodOptions): Promise<PaginatedResponse<Pod>> {
    const params = new URLSearchParams();
    if (options?.limit) params.set("limit", String(options.limit));
    if (options?.page_token) params.set("page_token", options.page_token);
    const qs = params.toString();
    return this.request("GET", `/pods${qs ? `?${qs}` : ""}`);
  }

  async get(podId: string): Promise<Pod> {
    return this.request("GET", `/pods/${podId}`);
  }

  async create(options: CreatePodOptions): Promise<Pod> {
    return this.request("POST", "/pods", options);
  }

  async delete(podId: string): Promise<void> {
    await this.request("DELETE", `/pods/${podId}`);
  }
}
